Classes/Service/NodeRedirectService.php

```diff
@@ -13,399 +13,216 @@
         }
 
-        if (method_exists(ActionRequest::class, 'fromHttpRequest')) {
-            $routeParameters = $httpRequest->getAttribute('routingParameters') ?? RouteParameters::createEmpty();
-            $httpRequest = $httpRequest->withAttribute('routingParameters', $routeParameters->withParameter('requestUriHost', $httpRequest->getUri()->getHost()));
-            // From Flow 6+ we have to use a static method to create an ActionRequest. Earlier versions use the constructor.
-            $this->actionRequestForUriBuilder = ActionRequest::fromHttpRequest($httpRequest);
-        } else {
-            /* @deprecated This case can be removed up when this package only supports Flow 6+. */
-            if ($httpRequest instanceof ServerRequest) {
-                $httpRequest = new \Neos\Flow\Http\Request([], [], [], [
-                    'HTTP_HOST' => $httpRequest->getHeaderLine('host'),
-                    'HTTPS' => $httpRequest->getHeaderLine('scheme') === 'https',
-                    'REQUEST_URI' => $httpRequest->getHeaderLine('path'),
-                ]);
-            }
-            $this->actionRequestForUriBuilder = new ActionRequest($httpRequest);
+        $affectedNode = $this->affectedNodes[$this->createAffectedNodesKey($nodeInfo, $contentRepositoryId)] ?? null;
+        if ($affectedNode === null) {
+            return;
         }
+        unset($this->affectedNodes[$this->createAffectedNodesKey($nodeInfo, $contentRepositoryId)]);
 
-        return $this->actionRequestForUriBuilder;
-    }
+        /** @var Uri $oldUri */
+        $oldUri = $affectedNode['url'];
+        $nodeType = $this->getNodeType($contentRepositoryId, $nodeInfo->getNodeTypeName());
 
-    /**
-     * Creates the queued redirects provided we can find the node.
-     *
-     * @return void
-     * @throws MissingActionNameException
-     */
-    public function createPendingRedirects(): void
-    {
-        if (!$this->enableAutomaticRedirects) {
+        if ($this->isRestrictedByNodeType($nodeType) || $this->isRestrictedByOldUri($oldUri->getPath())) {
             return;
         }
-
-        $this->nodeFactory->reset();
-        foreach ($this->pendingRedirects as $nodeIdentifierAndWorkspace => $oldUriPerDimensionCombination) {
-            [$nodeIdentifier, $workspaceName] = explode('@', $nodeIdentifierAndWorkspace);
-            $this->buildRedirects($nodeIdentifier, $workspaceName, $oldUriPerDimensionCombination);
+        try {
+            $newUri = $this->getNodeUriBuilder($nodeInfo->getSiteNodeName(), $contentRepositoryId)->uriFor($nodeAddress);
+        } catch (NoMatchingRouteException $exception) {
+            // We can't build an uri for given node, so we can't create any redirect. E.g.: Node is disabled.
+            return;
         }
-        $this->pendingRedirects = [];
+        $this->createRedirectWithNewTarget($oldUri->getPath(), $newUri->getPath(), $nodeInfo->getSiteNodeName());
 
         $this->persistenceManager->persistAll();
```

Comment:
Currently that `persistAll()` call breaks Neos Behat tests:
![image](https://github.com/neos/redirecthandler-neosadapter/assets/307571/de6dc044-234b-45ab-908e-165f4c7216ea)

But obviously that should not be the case.. Maybe we'll have to use separate connections instances for the projections etc.. I'll give it a try

Reply:
FYI: Using dedicated instances solves the problem for me but now tests fail due to some runtime cache:
![image](https://github.com/neos/redirecthandler-neosadapter/assets/307571/614ad89f-10c5-4824-80f0-4ee1ebecbe09)

(executing them individually works)

Reply:
I had similar issues i case one scenario was failing before (e.g disabled nodes)